src/main/java/org/marketdesignresearch/mechlib/mechanism/auctions/interactions/Interaction.java

```diff
@@ -0,0 +1,19 @@
+package org.marketdesignresearch.mechlib.mechanism.auctions.interactions;
+
+import org.marketdesignresearch.mechlib.core.bid.Bid;
+import org.marketdesignresearch.mechlib.core.bidder.Bidder;
+
+/**
+ * Also check activity rules here for submitted bids
+ * 
+ * @author Manuel Beyeler
+ *
+ * @param <T>
+ */
+public interface Interaction<T extends Bid> {
+	T proposeBid();
+	void submitBid(T bid);
+	Bidder getBidder();
+	
+	Class<? extends Interaction<T>> getType();
+}

```

Comment:
The new interaction interface. Note that this replaces the proposeBid and submitBid methods from the auction. An interaction is always bidder specific.

Reply:
Another solution would be to drop all Bid depending methods here and move them to the specific interaction only.

Reply:
I moved the methods away from this interface. Thus removing the generics.